molgenis-data-index/src/main/java/org/molgenis/data/index/IndexedRepositoryDecorator.java

```diff
@@ -135,4 +158,89 @@ private boolean querySupported(Query<Entity> q)
 		return !containsAnyOperator(q, unsupportedOperators) && !containsComputedAttribute(q, getEntityType())
 				&& !containsNestedQueryRuleField(q);
 	}
+
+	private long indexCount(Query<Entity> q)
+	{
+		try
+		{
+			return searchService.count(getEntityType(), q);
+		}
+		catch (UnknownIndexException e)
+		{
+			waitForIndexToBeStable();
+
+			// try again
+			try
+			{
+				return searchService.count(getEntityType(), q);
+			}
+			catch (UnknownIndexException e1)
+			{
+				throw translateUnknownIndexException();
+			}
+		}
+	}
+
+	private Object indexSearchOne(Query<Entity> q)
```

Comment:
The trying stuff twice can be put in single method (I've tried it out, will put up PR)